polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java

```diff
@@ -117,4 +122,16 @@ public interface PolarisConfigurationStore {
       return getConfiguration(ctx, config);
     }
   }
+
+  public static <T> @Nonnull T getConfiguration(PolarisConfiguration<T> configuration) {
+    CallContext callContext = CallContext.getCurrentContext();
+    if (callContext == null) {
+      return configuration.defaultValue;
```

Comment:
How is this code path engaged? When it is valid to use the default value in case `CallContext` is not available?

Reply:
To clarity: this change affects all config variables, even though the PR states it's adjusting the limit on storage config locations.

Reply:
When is it not valid to use the default value if CallContext it not available?

CallContext is only ever via `CallContext.getCurrentContext()` during tests, but my thinking was that it's outside the scope of this PR to go and fix all those tests. If you remove the check on line 128 and run the tests, you can see a handful of tests fail when they try to create storage configurations. If you'd prefer, I can try to fix those tests here.

It's not the case that this affects all config variables though, as no other config variable check currently uses this code path. Only the new one.

Reply:
>  the admin user changed the config.

From what I understand, the admin user can't do this in a situation where there is no callcontext. If there is no callcontext, this config check is not actually happening in the context of a call (ha) against a real service. It's only tests. If you want, I can go into those tests and try to wire up a CallContext threadlocal everywhere. I was approaching the problem with the mentality that we will try to remove that threadlocal soon anyway and this problem (and indeed this null-check) can go away soon.

Reply:
I do want to remove the ThreadLocal ASAP, but I wouldn't want to make refactoring away a ThreadLocal a blocker for a real feature that a user needs.

I am open to at least fix tests so that the check for a null callcontext can be removed if it wasn't needed. There are additional tests that have a null `PolarisCallContext` while having a non-null `CallContext` -- shall I fix those too? To me, it still seems that if there is no CallContext then implicitly there can be no overrides as we aren't in the process of a real call against a real service that has been configured by an admin like you said.

Reply:
I'll remove this method for now, but note the near-identical existing method that can be used in its place.